README.md

```diff
@@ -16,3 +17,7 @@ Also give a theoretical argument for why X could or could not be correct, based
 on the complexity of the general sorting problem we covered in class.
 
 Add your answers to this markdown file.
+
+## Response
+
+I would likely try using the blackbox implementation against all the permutations of several differently sized lists, and chart how long it takes to sort each permutation of a given list. I would test this claim by getting say 10 lists of size n and recording the algorithms runtime with those lists as inputs. Then I would repeat the process several times with lists of size Sn (where S is some positive scalar). If for each batch of lists of size Sn, $averageRuntime(listsOfSize(Sn)) \approx S*averageRuntime(listsOfSize(n))$ and the algorithm sorts all of the lists correctly, it would then be indicative of an $O(n)$ complexity. I would however be extremely skeptical of any sorting algorithm claiming to have $O(n)$ complexity. I would predict the charted runtimes to be fairly different for different permutations of a given list. The reason I would be skeptical of sorting algorithm claiming $O(n)$ complexity is because for every comparison based sorting algorithm, in order to sort every permutation of a list the binary tree tracking the comparisons (e.i. $arr[i] < arr[i+1] \implies left; else \implies right$ ) must have a depth of at least $\log(n!).$ This essentially means our worst case scenario of a comparison based sorting algorithm is bounded by $\Omega(n\log(n))$ In this case, $O(n)$ would violate this theoretical bound.
```

Comment:
How would you find this $S$?

Reply:
If f(x) returns the average runtime, and the input is list size, then a linear relationship would mean f(Sn) ≈ S * f(n) should be true for any positive scalar S. These values could be incremental (i.e. n, 2n, 3n, etc.) or random (0.5n, 10n, 35n), just so long as we can get a good scope of how it grows. I have attached a graph describing what a linear complexity would look like under this testing model, but please let me know if I'm not getting a hint. 

![image](https://github.com/COSC3020/theoretical-sorting-rmille70/assets/112435614/dade0aad-7651-4785-ab4b-f780e9c6e161)